kratos/input_output/vtk_output.h

```diff
@@ -0,0 +1,230 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Aditya Ghantasala
+//
+// 
+
+#if !defined(VTK_OUTPUT_PROCESS_H)
+#define VTK_OUTPUT_PROCESS_H
+// System includes
+#include <vector>
+#include <iostream>
+#include <string>
+#include <fstream>
+#include <algorithm>
+#include <iomanip> // for std::setprecision
+#include <map>
+#include "includes/kratos_parameters.h"
+#include "includes/io.h"
+// project includes
+
+namespace Kratos
+{
+/** \brief VtkOutput
+* A simple class that has functionality to write vtk output
+*/
+class KRATOS_API(KRATOS_CORE) VtkOutput : public IO
+{
+
+    typedef ProcessInfo ProcessInfoType;
+
+  public:
+    /// Pointer definition of VtkOutput
+    KRATOS_CLASS_POINTER_DEFINITION(VtkOutput);
+
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Constructor by passing a vector of Master and slave dofs and corresponding Matrix and constant vector
+     * @param rModelPart The modelpart which is used for output
+     * @param rParameters Parameters including settings for the output
+     */
+    VtkOutput(ModelPart &rModelPart, Parameters rParameters);
+    /// Destructor.
+    virtual ~VtkOutput();
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /**
+     * @brief Prints mrModelPart in VTK format together with the results
+     */
+    void PrintOutput();
+
+    ///@}
+
+    ///@name Access
+    ///@{
+
+    ///@
+
+    ///@name Static Operations
+    ///@{
+    void GetInfo() const;
+
+    ///@}
+    /**
+     * @brief Prints information about the class
+     * @param rOStream ostream object where output is printed
+     */
+    void PrintInfo(std::ostream &rOStream) const
+    {
+        rOStream << " VtkOutput object " << std::endl;
+    }
+
+    enum class FileFormat{
+        VTK_ASCII,
+        VTK_BINARY
+    };
+
+    enum class WriteDataType{
+        VTK_SCALAR,
+        VTK_VECTOR
+    };
+
+  private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+    ModelPart &mrModelPart;
+    std::string mCaseName;
+    std::string mOutputFilename;
+    VtkOutput::FileFormat mFileFormat;
+
+    Parameters mrOutputSettings;
+    unsigned int mDefaultPrecision;
+    std::map<int, int> mKratosIdToVtkId;
+    unsigned int mVtkCellListSize;
+    bool mDoneTest;
+    bool mShouldSwap;
```

Comment:
what are those members and do they have to be members?

Reply:
actually yes !